### ISSUE TITLE
pRIblast ruuning question

### ISSUE THREAD

**Issue by fanglongfa** [UDC-GAC/pRIblast#6]
mpirun -np 6 -x OMP_NUM_THREADS=6 pRIblast ris -i LncRNA.fa -o ZMpredictions.txt -d ZM -a area

[lzu-MZ72-HB0-00:2149410] *** Process received signal ***
[lzu-MZ72-HB0-00:2149410] Signal: Aborted (6)
[lzu-MZ72-HB0-00:2149410] Signal code:  (-6)
[lzu-MZ72-HB0-00:2149410] [ 0] /lib/x86_64-linux-gnu/libc.so.6(+0x42520)[0x14db7f442520]
[lzu-MZ72-HB0-00:2149410] [ 1] /lib/x86_64-linux-gnu/libc.so.6(pthread_kill+0x12c)[0x14db7f4969fc]
[lzu-MZ72-HB0-00:2149410] [ 2] /lib/x86_64-linux-gnu/libc.so.6(raise+0x16)[0x14db7f442476]
[lzu-MZ72-HB0-00:2149410] [ 3] /lib/x86_64-linux-gnu/libc.so.6(abort+0xd3)[0x14db7f4287f3]
[lzu-MZ72-HB0-00:2149410] [ 4] /mnt/sdb/fanglf/anaconda3/envs/CIRI/lib/libstdc++.so.6(+0xb135a)[0x14db7f8b135a]
[lzu-MZ72-HB0-00:2149410] [ 5] /mnt/sdb/fanglf/anaconda3/envs/CIRI/lib/libstdc++.so.6(+0xb13c5)[0x14db7f8b13c5]
[lzu-MZ72-HB0-00:2149410] [ 6] /mnt/sdb/fanglf/anaconda3/envs/CIRI/lib/libstdc++.so.6(+0xb1658)[0x14db7f8b1658]
[lzu-MZ72-HB0-00:2149410] [ 7] /mnt/sdb/fanglf/anaconda3/envs/CIRI/lib/libstdc++.so.6(_ZSt20__throw_length_errorPKc
[lzu-MZ72-HB0-00:2149410] [ 8] pRIblast(+0x35277)[0x557d90c24277]
[lzu-MZ72-HB0-00:2149410] [ 9] pRIblast(+0x1a10e)[0x557d90c0910e]
[lzu-MZ72-HB0-00:2149410] [10] pRIblast(+0x3029b)[0x557d90c1f29b]
[lzu-MZ72-HB0-00:2149410] [11] pRIblast(+0xee2a)[0x557d90bfde2a]
[lzu-MZ72-HB0-00:2149410] [12] /lib/x86_64-linux-gnu/libc.so.6(+0x29d90)[0x14db7f429d90]
[lzu-MZ72-HB0-00:2149410] [13] /lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0x80)[0x14db7f429e40]
[lzu-MZ72-HB0-00:2149410] [14] pRIblast(+0x11c05)[0x557d90c00c05]
[lzu-MZ72-HB0-00:2149410] *** End of error message ***

**Comment by amatria** [comment]
Hi :)

Can you please provide me with your input files so I can investigate the issue?

**Comment by fanglongfa** [comment]
OK， the attachment contains my test files, where Zhongmu.fa is used to construct the database file, and LncRNA.fa is the query file.
 thank you!







At 2023-10-18 00:03:46, "Iñaki Amatria Barral" ***@***.***> wrote:

Hi :)

Can you please provide me with your input files so I can investigate the issue.

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

从网易126邮箱发来的超大附件推荐客户端极速下载
LncRNA.fa (32.73M, 2023年11月2日 12:08 到期)
下载
Zhongmu.fa (61.79M, 2023年11月2日 12:08 到期)
下载

**Comment by amatria** [comment]
Hi,

I cannot download the attachments. You must manually upload them to the GitHub issue. However, I can see that your target FASTA file is 62MB in size, and that your query FASTA file is 33MB in size. So, this is likely an out of memory issue. Have you tried rebuilding the database using the parameter `-c <INT>` (i.e., database page size)? This information is sourced from the original [pRIblast paper](https://doi.org/10.1016/j.future.2022.08.014):

_4.2. Database paging_

_RIblast writes results to the output file only after all target sequences have been compared against a certain lncRNA query sequence. Therefore, if a target database is too large or a query segment produces a big amount of prediction candidates (i.e., seeds), memory may be filled up before results are finally saved to disk._

_pRIblast introduces a very effective optimization to overcome this limitation. It divides the target database into several smaller and independent pages, which, because of their reduced size, will not overflow the available memory space. Put differently, because results can only be written after a sequence has been compared against a target database, pRIblast makes predictions against subsets of such database. In this way, once a query sequence has been compared against all subsets of the target dataset, the output file will hold the same results as if the sequence had been compared against the whole target database in one single round._

_Because of the nature of the data structures stored in the database, it is not possible to paginate a target dataset with no previous preprocessing. Therefore, the database construction step (see Section [3](https://www.sciencedirect.com/science/article/pii/S0167739X22002758?via%3Dihub#sec3)) has been slightly modified to be able to read chunks of a fixed size later in the RNA interaction search step. In any case, databases built without the tweaked version of the construction step are backwards compatible with the new pRIblast workflow._

In simpler terms:
1. Rebuild your database using the `-c <INT>` parameter. For example, try `-c 250` (which processes the target sequences in chunks of 250).
```
$ mpirun -np 6 -x OMP_NUM_THREADS=6 ./target/pRIblast.release db -i Zhongmu.fa -o ZM-250 -c 250 -a block
```
2. Execute the RNA interaction search step again and monitor memory usage. If you still run out of memory, rebuild the database again, but this time with a lower chunk size value, such as `-c 100`. You can even choose a chunk size of 1 if memory is limited, but this will impact the tool's performance.
```
$ mpirun -np 6 -x OMP_NUM_THREADS=6 ./target/pRIblast.release ris -i LncRNA.fa -d ZM-250 -o ZMpredictions.txt -c 250 -a area
```

Furthermore, judging by the size of your inputs, it's likely that you'll require ample disk space to store the output interactions resulting from your analysis. Refer to the table below, once more obtained from the original [pRIblast paper](https://doi.org/10.1016/j.future.2022.08.014), which can provide you with an estimate of the potential growth in the size of output files (Anser is approximately 12MB in size):
<p><img width="720" src="https://github.com/UDC-GAC/pRIblast/assets/34030065/f9df135d-8b73-47ec-8f91-0b3fa33d6cce"></p>

I hope this helps :-).
Iñaki

**Comment by fanglongfa** [comment]
ok, thank you, I will try it.
